Add new select component for <StateSortedSchools> scene - #210

jjhampton opened the pull request:
# Description of changes
Implements new select component for #210 using the [react-selec](https://github.com/JedWatson/react-select)t library.  On the Code School page 'Schools By State' section at the bottom, users can:
* Display the state options by clicking the down caret in the Select component (like a traditional HTML select element)
* Type in the state names in the Select component.  This filters the state options displayed, based on the search query

When state options are displayed:
* Arrow keys + enter can be used to navigate the list and select a state
* Mouse pointer clicks can also be used to select a state
* Multiple states can be added to a selection
* Individual states can be cleared from the selection
* All states can be cleared at once from the selection

School cards are displayed on the screen below the Select component, based on which states are selected.  When multiple states are selected, the school cards are sorted alphabetically by state.

# Issue Resolved
Closes #210

## Notes
The Select component will now be available for use on other pages via `import Select from 'react-select';`

To import the global CSS styles for the Select component from the react-select dependency, I created src/shared/styles/react-select.global.css (similar to what we are already doing for react-table and normalize.css).  But, unlike with those files, I used the CSS `@import` syntax within react-select.global.css to import the CSS styles from the dependency's styles in the node_modules directory.  We may want to adopt a similar pattern for react-table or normalize, instead of hardcoding in those CSS styles into our app.

kylemh commented:
Everything looks really great, on all mobile browsers for iOS, the client zoomed in when the Select2 component was selected. Does this occur on Android as well? I'm willing to approve this PR anyways, but I believe that would be the only improvement I can think of.

jjhampton commented:
@kylemh Good call to check how the mobile UI looks - I totally forgot about that. Checked it on Chrome for Android, didn't experience any zooming, worked pretty much same as desktop Chrome.  We can always open another mobile-only issue to fix that though, if we want to progress incrementally.

kylemh commented:
That SO thread is the exact user story I experienced for clear reference - definitely follow that.

kylemh commented:
Since you're going back in there, there are 2 minor style changes I would implement:

1) Separate the clear button a little farther from the down-arrow icon, and make it larger.
2) Make the entire component around 15% larger - text and all.

jjhampton commented:
@kylemh Added some fixes in response to code review: 
* added fix to prevent zoom for iOS (tested on Safari, but not iOS Chrome)
* added some overrides for to global react-select CSS to increase height/line-height/font-sizes where possible 
* set the width of the Select (within the StateSortedSchools component) to full-width of the section, but with a max-width

We can probably tweak these styles more later when we add some app-wide styles for Form Components.
